apps/web/app/(navigation)/LanguageSelector.tsx

```diff
@@ -0,0 +1,60 @@
+import { languages } from "lib/contexts";
+import * as Select from "@radix-ui/react-select";
+
+import { Flag } from "ui";
+import {
+  SelectContentStyled,
+  SelectItemStyled,
+  SelectTriggerStyled,
+} from "./LanguageSelectorStyled";
+import { useEffect } from "react";
+import { languageAtom } from "app/store";
+import { useSetAtom, useAtomValue } from "jotai";
+import { useHasScrollBar } from "lib/hooks/useHasScrollBar";
+
+export const LanguageSelector = () => {
+  const { hasScrollbar } = useHasScrollBar();
+  const setLanguage = useSetAtom(languageAtom);
+  const language = useAtomValue(languageAtom);
+
+  useEffect(() => {
+    const lang = localStorage.getItem("lang");
+    lang && setLanguage((localStorage.getItem("lang") as languages) || "en");
+  }, [setLanguage, language]);
+
+  const changeLanguage = (lang: string) => {
+    setLanguage(lang as languages);
+    localStorage.setItem("lang", lang);
+  };
+
+  const items = [
+    { lang: "pl", flagSrc: "/flags/pl.svg", languageName: "Polski" },
+    { lang: "en", flagSrc: "/flags/en.svg", languageName: "English" },
+    { lang: "fr", flagSrc: "/flags/fr.svg", languageName: "Français" },
+  ];
+
+  return (
+    <Select.Root
+      value={(language as languages) || "en"}
+      onValueChange={(lang) => {
+        changeLanguage(lang as languages);
+      }}>
```

Comment:
you don't need casting here
```suggestion
    <Select.Root
      value={language || "en"}
      onValueChange={(lang) => {
        changeLanguage(lang);
      }}>
```

Reply:
Changed